diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LineLength.kt

```diff
@@ -74,6 +123,161 @@ class LineLength : Rule("line-length") {
         }
     }
 
+    private fun fixError(wrongNode: ASTNode, configuration: LineLengthConfiguration) {
+        var parent = wrongNode
+        do {
+            when (parent.elementType) {
+                FUN -> {
+                    if (!parent.hasChildOfType(BLOCK))
+                        parent.addChild(PsiWhiteSpaceImpl("\n"), parent.findChildByType(EQ)!!.treeNext)
+                    return
+                }
+                CONDITION -> {
+                    fixCondition(parent, configuration)
+                    return
+                }
+                PROPERTY -> {
+                    fixProperty(parent, configuration)
+                    return
+                }
+                EOL_COMMENT -> {
+                    fixComment(wrongNode, configuration)
+                    return
+                }
+                else -> parent = parent.treeParent
+            }
+        } while (parent.treeParent != null)
+    }
+
+    private fun fixComment(wrongNode: ASTNode, configuration: LineLengthConfiguration) {
+        val leftOffset = positionByOffset(wrongNode.startOffset).second
+        val indexLastSpace = wrongNode.text.substring(0, configuration.lineLength.toInt() - leftOffset).lastIndexOf(' ')
+        if (indexLastSpace == -1)
+            return
+        val nodeText = "//${wrongNode.text.substring(indexLastSpace, wrongNode.text.length)}"
+        wrongNode.treeParent.run {
+            addChild(LeafPsiElement(EOL_COMMENT, wrongNode.text.substring(0, indexLastSpace)), wrongNode)
+            addChild(PsiWhiteSpaceImpl("\n"), wrongNode)
+            addChild(LeafPsiElement(EOL_COMMENT, nodeText), wrongNode)
+            removeChild(wrongNode)
+        }
+    }
+
+    private fun fixCondition(wrongNode: ASTNode, configuration: LineLengthConfiguration,
```

Comment:
I think this method should be renamed, because it is used not only for conditions. Also, it has become complex enough to deserve KDoc with description of parameters.